tests/test_cpu_throttle/assert

```diff
@@ -8,92 +8,99 @@ import os
 
     def test_all(self):
-        self.set_cfs_quota('50000')
+        self.install_module()
         self.init_cgroup()
+        self.set_cfs_quota('50000')
         self.check_le_75()
         self.check_after_load()
         self.set_cfs_quota('100000')
         self.check_gt_75()
         self.check_after_unload()
-        
+
     def check_le_75(self):
         cpu_util = self.get_cpu_util(self.child.pid)
-        # assert cpu_util <= 75
-        if cpu_util > 75:
-            self.error_handler(0, 75)
+        self.validate_lt(cpu_util, 75)
 
-    def check_after_load(self):
+    def install_module(self):
         scheduler_rpm = glob(os.path.join('/tmp/work', 'scheduler*.rpm'))
         if len(scheduler_rpm) != 1:
             print("Please check your scheduler rpm");
             self.teardown_class()
             sys.exit(1)
         scheduler_rpm = scheduler_rpm[0]
         sh.rpm('-ivh', scheduler_rpm)
+        sh.tee(sh.echo(0), '/sys/kernel/plugsched/plugsched/enable')
+
+    def uninstall_module(self):
+        sh.rpm('-e', 'scheduler-xxx')
+
+    def check_after_load(self):
+        sh.tee(sh.echo(1), '/sys/kernel/plugsched/plugsched/enable')
         cpu_util = self.get_cpu_util(self.child.pid)
-        # assert cpu_util <= 75
-        if cpu_util > 75:
-            self.error_handler(0, 75)
+        self.validate_lt(cpu_util, 75)
 
     def check_gt_75(self):
         cpu_util = self.get_cpu_util(self.child.pid)
-        # assert cpu_util >= 75
         if cpu_util < 75:
             self.error_handler(1, 75)
 
     def check_after_unload(self):
-        sh.rpm('-e', 'scheduler-xxx')
+        sh.tee(sh.echo(0), '/sys/kernel/plugsched/plugsched/enable')
         cpu_util = self.get_cpu_util(self.child.pid)
-        # assert cpu_util >= 75
-        if cpu_util < 75:
-            self.error_handler(1, 75)
+        self.validate_gt(cpu_util, 75)
 
     def get_cpu_util(self, pid):
-        def cpu_usage():
-            time.sleep(0.5)
-            herts = float(sh.getconf('CLK_TCK'))
-            process_file = "/proc/" + str(pid) + "/stat"
-            process_stat = sh.cat(process_file).split()
-            total_time = float(process_stat[13]) + float(process_stat[14])
-            elapsed_seconds = time.time() - self.start_time
-            return 100.0 * ((total_time / herts) / elapsed_seconds)
-        
-        cpu_util_1 = cpu_usage()
-        cpu_util_2 = cpu_usage()
-        cpu_util = (cpu_util_1 + cpu_util_2) / 2
-        return cpu_util
+        interval = 2
+        herts = float(sh.getconf('CLK_TCK'))
+        process_file = "/proc/" + str(pid) + "/stat"
+        process_stat = sh.cat(process_file).split()
+        total_time1 = float(process_stat[13]) + float(process_stat[14])
+        time.sleep(interval)
+        process_stat = sh.cat(process_file).split()
+        total_time2 = float(process_stat[13]) + float(process_stat[14])
+        return 100.0 * (total_time2 - total_time1) / (herts * interval)
```

Comment:
Can we use the python psutil library? It is easy to use, just like following:

import psutil
p = psutil.Process(pid)
util = p.cpu_percent(interval=1)

Reply:
Great